src/display/pattern_helper.js

```diff
@@ -85,42 +90,53 @@ class RadialAxialShadingPattern extends BaseShadingPattern {
     return grad;
   }
 
-  getPattern(ctx, owner, inverse, shadingFill = false) {
+  getPattern(ctx, owner, inverse, pathType) {
     let pattern;
-    if (!shadingFill) {
-      if (this.cachedCanvasPatterns.has(this)) {
-        pattern = this.cachedCanvasPatterns.get(this);
-      } else {
-        const tmpCanvas = owner.cachedCanvases.getCanvas(
-          "pattern",
-          owner.ctx.canvas.width,
-          owner.ctx.canvas.height,
-          true
-        );
-
-        const tmpCtx = tmpCanvas.context;
-        tmpCtx.clearRect(0, 0, tmpCtx.canvas.width, tmpCtx.canvas.height);
-        tmpCtx.beginPath();
-        tmpCtx.rect(0, 0, tmpCtx.canvas.width, tmpCtx.canvas.height);
+    if (pathType === PathType.STROKE || pathType === PathType.FILL) {
+      const ownerBBox = owner.current.getClippedPathBoundingBox(
+        pathType,
+        ctx.mozCurrentTransform
+      );
```

Comment:
Looking at the previous test-results, there's also a breaking error which I assume is caused by this line (since this method uses `Util.intersect` which can return `null`).
```
TEST-UNEXPECTED-FAIL | test failed issue6961 | in firefox | page1 round 1 | render : TypeError: can't access property 2, ownerBBox is null
```

Reply:
Nice catch, I totally missed that! It seems to fix the other issues. However, that PDF is now displaying wrong in a new way (it's wrong as is too). The computed bounding box for the shading is slightly too small. I'm not sure why yet. It's not going to be fun to debug, since that PDF has thousands of paths. :( 

I still don't quite understand why breaking that PDF, breaks other PDFs. We shouldn't be reusing any canvases, but we must not be cleaning up something correctly when an error happens.

Reply:
Could the problem be that we're re-using the *same* `<canvas>` for every page/document during tests, thus possibly keeping some bad graphics-state alive?

A while back I wanted to see if we could get rid of some intermittent failures by always creating a new `<canvas>` during tests, please see PR #12960; maybe we should see if that'd fixed these problems and potentially land that?

Reply:
That brings up a good point! The canvas state should be getting reset when we assign to the width/height in driver.js, but in chrome we add our polyfill for mozCurrentTransform and that does NOT get reset. That would explain why this was only an issue in chrome. I'm thinking we should either override the width/height attributes of the canvas to reset the transform stuff or manually wipe out the transform stack and reset the current transform. The former is probably less error prone since we wouldn't have to remember to do it.

Reply:
As for the bounding box being wrong in issue6961, that is because of #14241.